src/core/lib/surface/call.cc

```diff
@@ -1851,9 +1851,8 @@ static grpc_call_error call_start_batch(grpc_call* call, const grpc_op* ops,
         {
           grpc_error* override_error = GRPC_ERROR_NONE;
           if (op->data.send_status_from_server.status != GRPC_STATUS_OK) {
-            override_error =
```

Comment:
Can we keep this part?  It looks like just the test is flaking, and this change hasn't caused any other tests to fail.

Reply:
Sure. I wonder how to do that on a revert PR... would it be too much trouble to have this revert go in as-is and then introduce this change? Otherwise this PR wouldn't really be a revert and things would be confusing.

Reply:
That's fine